### PR TITLE
Added support for AWS

### PR DESCRIPTION
**Pull request by oneilcin**
Supports AWS (as well as MaaS) now.  Added a sample yaml file in assets.

### CODE REVIEW

**Comment by leahnp** [comment]
I will help look into CI on this one - seems we need to modify our Dockerfile.

File: pkg/apis/clustercontroller/v1alpha1/types.go [hunk 1]
```diff
@@ -19,7 +19,8 @@ type KrakenCluster struct {
 type CloudProviderCredentials struct {
 	Username   string `json:"username,omitempty"`
 	Password   string `json:"password,omitempty"`
-	SecretName string `json:"secretname,omitempty"`
+	Accesskey  string `json:"accesskey,omitempty"`
```
Comment (DStorck) [inline pkg/apis/clustercontroller/v1alpha1/types.go]: accessKey needs Key to be capitalized

**Comment by DStorck** [comment]
Seems like we also need to add CloudProvider field for Region

File: pkg/apis/clustercontroller/v1alpha1/types.go (hunk 1 above)
Comment (oneilcin) [reply pkg/apis/clustercontroller/v1alpha1/types.go]: In main.go, line 369, Here is how I use your structure ```cluster := &gogo.Juju{
			Name:   string(kc.UID),
			Kind:   gogo.Aws,
			Bundle: JujuBundle,
			AwsCr: gogo.AWSCredentials{
				Username:  kc.Spec.CloudProvider.Credentials.Username,
				AccessKey: kc.Spec.CloudProvider.Credentials.Accesskey,
				SecretKey: kc.Spec.CloudProvider.Credentials.Password,
			},
			AwsCl: gogo.AWSCloud{
				Region: kc.Spec.CloudProvider.Region,
			},```

File: rootfs/cluster-controller/Dockerfile [hunk 2]
```diff
@@ -49,15 +52,8 @@ RUN mkdir /bin/controller
...
-RUN go get github.com/sirupsen/logrus
-RUN git clone https://github.com/dstorck/gogo.git /bin/gogo
-RUN cd /bin/gogo && \
-		go get github.com/dstorck/gogo
+COPY --from=0 /app/main /bin/controller
 
 # ENTRYPOINT /bin/controller/main -kubeconfig=""
-CMD ["/bin/controller/main"]
+#CMD ["/bin/controller/main"]
+
```
Comment (oneilcin) [inline rootfs/cluster-controller/Dockerfile]: @leahnp I left CMD commented (line 58) when I tested with a deployment.  Do you want me to uncomment it?

Comment (leahnp) [reply rootfs/cluster-controller/Dockerfile]: I can do this with my Helm work tomorrow

Comment (oneilcin) [reply rootfs/cluster-controller/Dockerfile]: Cool, ready to merge!!

Comment (oneilcin) [reply rootfs/cluster-controller/Dockerfile]: Also tomorrow check /app/main, does that dir exist now?

Comment (oneilcin) [reply rootfs/cluster-controller/Dockerfile]: It must have because main got copied correctly.

**Pull request merged**